lox_services/persistence/database/query_handlers.py

```diff
@@ -47,40 +57,54 @@ def raw_query(query: str, print_query: bool = True) -> QueryJob:
         ))
 
     bigquery_client = Client()
-    query_job = bigquery_client.query(query)
+    
+    if parameters: 
+        parameters = QueryJobConfig(
+        query_parameters=[
+            ArrayQueryParameter(parameter[0], parameter[1].value, parameter[2]) 
+            if (isinstance(parameter[2], Sequence) and not isinstance(parameter[2], str)) 
+            else ScalarQueryParameter(parameter[0], parameter[1].value, parameter[2])  for parameter in parameters
+        ])
+    
+    query_job = bigquery_client.query(query, job_config=parameters)
     query_job.result()
     return query_job
 
 
-def select(query: str, print_query: bool = True) -> DataFrame:
+def select(query: str, print_query: bool = True, *, parameters:  Optional[Sequence[Tuple[str, BQParameterType, Any]]] = None) -> DataFrame:
     """Checks if the query begings with a SELECT statement. If so the query is being executed.
         ## Arguments
         - `query`: String representation of the query to be executed.
         - `print_query`: Tells whether or not to print the query before executing it.
+        - `parameters`: list of parameter #used to avoid sql injection
 
         ## Example
-            >>> select("SELECT * FROM InvoicesData.Refunds LIMIT 10")
+            >>> select("SELECT * FROM InvoicesData.Refunds where carrier="UPS" LIMIT 10")
+            >>> select ("SELECT * FROM InvoicesData.Refunds where carrier=@carrier LIMIT 10", parameters = [("carrier", "STRING", "UPS")])
 
         ## Return
         The result of the select query as a dataframe.
     """
     if not (query.lstrip().startswith("SELECT") or query.lstrip().startswith("WITH")) :
         raise BadQueryTypeException('SELECT or WITH')
 
-    return raw_query(query, print_query).result().to_dataframe()
+    return raw_query(query, print_query = print_query, parameters = parameters).result().to_dataframe()
 
 
-def update(query: str, print_query: bool = True) -> int:
+def update(query: str, print_query: bool = True, *, parameters:  Optional[Sequence[Tuple[str, BQParameterType, Any]]] = None) -> DataFrame:
     """Checks if the query begings with a UPDATE statement. If so the query is being executed.
         ## Arguments
         - `query`: String representation of the query to be executed.
         - `print_query`: Tells whether or not to print the query before executing it.
+        - `parameters`: list of parameter #used to avoid sql injection
 
         ## Example
             >>> update("UPDATE InvoicesData.Refunds SET state='Test' WHERE company='Test'")
+            >>> update("UPDATE InvoicesData.Refunds SET state='Test' where company=@company", parameters = [("company", "STRING", "Test")])
```

Comment:
If your type hints are correct, then this code would not work.

`parameters = [("company", BQParameterType.STRING, "Test")])`